libc: fails to build with `add_custom_target cannot create target "libc" because another target with the same name already exists`

Not sure what caused it. Maybe #117220  by @nickdesaulniers 
```
-- Looking for logf128 - found
-- Targeting M68k
-- Targeting Xtensa
-- Targeting AArch64
-- Setting LIBC_NAMESPACE namespace to '__llvm_libc_20_0_0_'
g++: error: unrecognized command-line option '--print-resource-dir'; did you mean '--print-search-dirs'?
g++: fatal error: no input files
compilation terminated.
-- Set COMPILER_RESOURCE_DIR to /usr/lib/gcc/aarch64-linux-gnu/10/ using --print-search-dirs
-- Building libc for aarch64 on linux with
        LIBC_COMPILE_OPTIONS_DEFAULT: 
-- LIBC_CONF_ENABLE_STRONG_STACK_PROTECTOR: ON
```

Regression range 356df2dd72e8..1e7171f692d0
Fails log:
https://llvm-jenkins.debian.net/job/llvm-toolchain-bullseye-binaries/architecture=arm64,distribution=bullseye,label=arm64/1608/console

<details>
<summary>actual cmake error</summary>

```
CMake Error at cmake/modules/LLVMExternalProjectUtils.cmake:453 (add_custom_target):
  add_custom_target cannot create target "libc" because another target with
  the same name already exists.  The existing target is a static library
  created in source directory "/build/source/libc/lib".  See documentation
  for policy CMP0002 for more details.
Call Stack (most recent call first):
  runtimes/CMakeLists.txt:266 (llvm_ExternalProject_Add)
  runtimes/CMakeLists.txt:559 (runtime_default_target)


CMake Error at cmake/modules/LLVMExternalProjectUtils.cmake:453 (add_custom_target):
  add_custom_target cannot create target "install-libc" because another
  target with the same name already exists.  The existing target is a custom
  target created in source directory "/build/source/libc/lib".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  runtimes/CMakeLists.txt:266 (llvm_ExternalProject_Add)
  runtimes/CMakeLists.txt:559 (runtime_default_target)


CMake Error at cmake/modules/LLVMExternalProjectUtils.cmake:453 (add_custom_target):
  add_custom_target cannot create target "install-libc-stripped" because
  another target with the same name already exists.  The existing target is a
  custom target created in source directory "/build/source/libc/lib".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  runtimes/CMakeLists.txt:266 (llvm_ExternalProject_Add)
  runtimes/CMakeLists.txt:559 (runtime_default_target)


CMake Error at cmake/modules/LLVMExternalProjectUtils.cmake:453 (add_custom_target):
  add_custom_target cannot create target "check-libc" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory "/build/source/libc/test".  See documentation
  for policy CMP0002 for more details.
Call Stack (most recent call first):
  runtimes/CMakeLists.txt:266 (llvm_ExternalProject_Add)
  runtimes/CMakeLists.txt:559 (runtime_default_target)
```
</details>

--- a/libc/CMakeLists.txt
+++ b/libc/CMakeLists.txt
@@ -52,22 +52,9 @@ set(LIBC_NAMESPACE ${default_namespace}
 
 # We will build the GPU utilities if we are not doing a runtimes build.
 option(LIBC_BUILD_GPU_LOADER "Always build the GPU loader utilities" OFF)
-if(LIBC_BUILD_GPU_LOADER OR NOT LLVM_RUNTIMES_BUILD)
-  foreach(_name ${LLVM_RUNTIME_TARGETS})
-    if("libc" IN_LIST RUNTIMES_${_name}_LLVM_ENABLE_RUNTIMES)
-      if("${_name}" STREQUAL "amdgcn-amd-amdhsa" OR "${_name}" STREQUAL "nvptx64-nvidia-cuda")
-        set(LIBC_NEED_LOADER_UTILS TRUE)
-      endif()
-    endif()
-  endforeach()
-  if("${LIBC_TARGET_TRIPLE}" STREQUAL "amdgcn-amd-amdhsa" OR
-     "${LIBC_TARGET_TRIPLE}" STREQUAL "nvptx64-nvidia-cuda")
-    set(LIBC_NEED_LOADER_UTILS TRUE)
-  endif()
-  if(LIBC_NEED_LOADER_UTILS)
-    add_subdirectory(utils/gpu)
-    return()
-  endif()
+if(LIBC_BUILD_GPU_LOADER OR ((NOT LLVM_RUNTIMES_BUILD) AND LLVM_LIBC_GPU_BUILD))
+  add_subdirectory(utils/gpu)
+  return()
 endif()
 
 add_subdirectory(newhdrgen)

--- a/llvm/cmake/modules/CrossCompile.cmake
+++ b/llvm/cmake/modules/CrossCompile.cmake
@@ -78,6 +78,9 @@ function(llvm_create_cross_target project_name target_name toolchain buildtype)
       list(APPEND libc_flags -DLLVM_FORCE_BUILD_RUNTIME=ON)
     endif()
   endif()
+  if(LLVM_LIBC_GPU_BUILD)
+    list(APPEND libc_flags -DLLVM_LIBC_GPU_BUILD=ON)
+  endif()
 
   add_custom_command(OUTPUT ${${project_name}_${target_name}_BUILD}/CMakeCache.txt
     COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}"

